### ISSUE TITLE
Sh Helper vs ShAction

### ISSUE THREAD

**Issue by raymondhoagland** [fastlane/fastlane#15016]
### Question Checklist

- [x] Updated fastlane to the latest version
- [x] I read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I read [docs.fastlane.tools](https://docs.fastlane.tools)
- [x] I searched for [existing GitHub issues](https://github.com/fastlane/fastlane/issues)

### Question Subject
Confused about the [sh helper function](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/fast_file.rb#L194-L215) specific to the Fastfile.  Is there a reason this doesn't use [ShAction](https://github.com/fastlane/fastlane/blob/master/fastlane/lib/fastlane/actions/sh.rb)?  This blocks the use of a step name (in one of the places I'd most like to have a description of what's happening).  If this _is_ a requirement, is the only way around that to directly call `other_action.ShAction`?  Would it be reasonable to add step_name as an accepted parameter for the helper function in the interim?

#### Question Description

```
lane :test do
  sh(command: 'ls', step_name: 'aa')
end
```
errors with `unknown keyword: step_name`.

Calling the ShActtion itself (e.g. from a Fastlane plugin) allows the use of step_name as it's calling the actual action which inherits step_name handling.

**Comment by janpio** [comment]
Step names are a pretty recent addition, so this might just have been missed back when they were added. @joshdholtz recently touched that code, so he might be able to say more about it.

**Comment by raymondhoagland** [comment]
@janpio I guess my concern is more that it doesn't seem to be calling ShAction at all - it's calling a separate helper function.  From what I can tell about the step name additions, it only applies to actions.

**Comment by janpio** [comment]
Yes, the `ShAction` is actually just for documentation and the actual `sh` is implemented via the `sh_helper`. So `sh` definitely is a special thing. It would probably be nicer if the code would be routed through the action, which could then still call the helper for execution - that would probably also allow step names and similar functionality to work.

If you want to tackle this, feel more than welcome to open a pull request.

**Comment by dotdoom** [comment]
@janpio I'm happy to send a PR. Do you think it would make sense to route it through [`CommandExecutor`](https://github.com/fastlane/fastlane/blob/d7e02e48e74c754d9d9194025815324d7380a479/fastlane_core/lib/fastlane_core/command_executor.rb) to have common code for running commands?

**Comment by janpio** [comment]
To be honest I have no idea @dotdoom - I would have to get an overview first what is going on there. I guess you already did that, so I will trust your judgement.
